test-harness/Makefile

```diff
@@ -1,18 +1,25 @@
+# Use bash shell with pipefail option enabled so that the return status of a
+# piped command is the value of the last (rightmost) commnand to exit with a
+# non-zero status. This lets us pipe output into tee but still exit on test
+# failures.
+SHELL = /bin/bash
+.SHELLFLAGS = -o pipefail -c
+
 all: test
 
 # wait for the device to be available before beginning the test
 # A temporary volume is mounted at /build when 'make test' is executing.
 # The following steps copy across useful output to this volume which can
 # then be extracted to form the CI summary for the test procedure.
 test:
-	retry -- tango_admin --check-device ska/basedevice/01
-	retry -- tango_admin --check-device ska/alarmhandler/01
-	retry -- tango_admin --check-device ska/capability/01
-	retry -- tango_admin --check-device ska/logger/01
-	retry -- tango_admin --check-device ska/master/01
-	retry -- tango_admin --check-device ska/obsdevice/01
-	retry -- tango_admin --check-device ska/subarray/01
-	retry -- tango_admin --check-device ska/telstate/01
+	retry --max=10 -- tango_admin --ping-device ska/basedevice/01
```

Comment:
Pipeline fails with the ping commands.

Reply:
if the pipeline fails when pinging the TMC devices, then the PR is working correctly! It indicates that one or more TMC devices could not be started in container, so there's either a problem with the device code or with the container setup.